### ISSUE TITLE
Allow attribute definition in any order

### ISSUE THREAD

**Issue by joeldrapper** [joeldrapper/literal#10]
Waiting on #11

**Comment by hangsu** [comment]
We ran into this:

```ruby
class A
  extend Literal::Attributes
  attribute :proc, Proc, :&
end

class B < A
  attribute :foobar, String
end
```

```
/Users/hangsu/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/literal-455f225574db/lib/literal/attributable.rb:68:in `module_eval': /Users/hangsu/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/literal-455f225574db/lib/literal/attributable.rb:71: syntax error, unexpected ',', expecting ')' (SyntaxError)
public def initialize(&proc, foobar:)
```

Anything we can do to help?

**Comment by joeldrapper** [comment]
This will be fixed in #83, which is almost ready.
